pthread_trace.cc

```diff
@@ -304,14 +304,31 @@ const char* getenv_or(const char* env, const char* def) {
 std::atomic<bool> initializing{false};
 std::atomic<bool> initialized{false};
 
+const char* get_trace_path() {
+  const char* path_ptr = getenv_or("PTHREAD_TRACE_PATH", "pthread_trace.proto");
+  if (getenv("IN_PTHREAD_TRACE_TEST")) {
```

Comment:
Rather than checking a new environment variable, I think this should just check if the path is an absolute path:

```
if (path_ptr[0] == '/') {
  return path_ptr;
}
```

The tests use `mktemp`, so this check will avoid prepending `TEST_UNDECLARED_OUTPUTS_DIR`, and it will also avoid messing with other paths that people make absolute.

(Also, `IN_PTHREAD_TRACE_TEST` isn't set anywhere in this PR, but rather than fix that, I suggest this fix instead.)